pkg/ccl/backupccl/tenant_backup_nemesis_test.go

```diff
@@ -0,0 +1,470 @@
+	require.NoError(t, err)
+	_, err = tenant10Conn.Exec("USE bank")
+	require.NoError(t, err)
+
+	// Import initial data
+	const numAccounts = 1000
+	const payloadSize = 100
+	const splits = 0
+	bankData := bank.FromConfig(numAccounts, numAccounts, payloadSize, splits)
+	l := workloadsql.InsertsDataLoader{BatchSize: 1000, Concurrency: 4}
+	_, err = workloadsql.Setup(ctx, tenant10Conn, bankData, l)
+	require.NoError(t, err)
+
+	backupLoc := "nodelocal://0/tenant-backup"
+
+	backupDone := make(chan struct{})
+	g := ctxgroup.WithContext(ctx)
+	g.GoCtx(func(ctx context.Context) error {
+		pgURL, cleanupGoDB, err := sqlutils.PGUrlE(
+			tenant10.SQLAddr(), "workload-worker" /* prefix */, url.User(username.RootUser))
+		if err != nil {
+			return err
+		}
+		defer cleanupGoDB()
+		reg := histogram.NewRegistry(20*time.Second, "bank")
+		ops, err := bankData.(workload.Opser).Ops(ctx, []string{pgURL.String()}, reg)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if ops.Close != nil {
+				ops.Close(ctx)
+			}
+		}()
+		fn := ops.WorkerFns[0]
+		for {
+			if err := fn(ctx); err != nil {
+				return err
+			}
+			select {
+			case <-backupDone:
+				return nil
+			default:
+			}
+		}
+	})
+
+	rng, _ := randutil.NewPseudoRand()
+	nemesisRunner := newBackupNemesis(t, rng, tenant10Conn)
+	nemesisRunner.Start(ctx)
+
+	var tablesToCheck []string
+	g.GoCtx(func(ctx context.Context) error {
+		defer close(backupDone)
+		defer nemesisRunner.Stop()
+
+		// Full Backup
+		done := nemesisRunner.RequireStart()
+		t.Logf("backup-nemesis: full backup started")
+		hostSQLDB.Exec(t, fmt.Sprintf("BACKUP TENANT 10 INTO '%s'", backupLoc))
+		t.Logf("backup-nemesis: full backup finished")
+		<-done
+
+		numIncrementals := 2 // rng.Intn(5) + 1
+		for i := 0; i < numIncrementals; i++ {
+			<-nemesisRunner.RequireStart()
+			tablesToCheck = nemesisRunner.TablesToCheck()
+
+			t.Logf("backup-nemesis: incremental backup started")
+			hostSQLDB.Exec(t, fmt.Sprintf("BACKUP TENANT 10 INTO LATEST IN '%s'", backupLoc))
+			t.Logf("backup-nemesis: incremental backup finished")
+		}
+		return nil
+	})
+
+	require.NoError(t, g.Wait())
+	hostSQLDB.Exec(t, fmt.Sprintf("RESTORE TENANT 10 FROM LATEST IN '%s' WITH tenant = '11'", backupLoc))
+
+	tenant11, err := tc.Servers[0].StartTenant(ctx, base.TestTenantArgs{
+		TenantID:            roachpb.MustMakeTenantID(11),
+		DisableCreateTenant: true,
+	})
+	require.NoError(t, err)
+
+	tenant11Conn, err := serverutils.OpenDBConnE(
+		tenant11.SQLAddr(), "bank", false, tenant11.Stopper())
+	require.NoError(t, err)
+
+	tenant10SQLDB := sqlutils.MakeSQLRunner(tenant10Conn)
+	tenant11SQLDB := sqlutils.MakeSQLRunner(tenant11Conn)
+	endTimeQuery := fmt.Sprintf(`SELECT max(end_time)::string FROM [SHOW BACKUP LATEST IN '%s']`, backupLoc)
+	endTime := hostSQLDB.QueryStr(t, endTimeQuery)[0][0]
+	assertEqualAcrossTenantRestore(t, tenant10SQLDB, tenant11SQLDB, "SELECT * FROM [SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE bank.bank]", endTime)
```

Comment:
I'm seeing failures that either indicate that we have real problems with incremental tenant backups, or that I'm simply picking the wrong timestamp here.  Does this look like a reasonable way to verify the restored tenant?

Reply:
Quick follow-up note, is that I've also run this check by using the AOST time on both the old and new tenant and hit issues.